src/ui/i18n/i18n.js

```diff
@@ -0,0 +1,136 @@
+import path from 'path';
+import Promise from 'bluebird';
+import { readFile } from 'fs';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+function getLocaleFromFileName(fullFileName) {
+  if (_.isEmpty(fullFileName)) throw new Error('Filename empty');
+
+  const fileExt = path.extname(fullFileName);
+  if (fileExt.length <= 0 || fileExt !== TRANSLATION_FILE_EXTENSION) {
+    throw new Error('Translations must be in a JSON file. File being registered is ' + fullFileName);
+  }
+
+  return path.basename(fullFileName, TRANSLATION_FILE_EXTENSION);
+}
+
+function getBestLocaleMatch(languageTag, registeredLocales) {
+  if (_.contains(registeredLocales, languageTag)) {
+    return languageTag;
+  }
+
+  // Find the first registered locale that begins with one of the language codes from the provided language tag.
+  // For example, if there is an 'en' language code, it would match an 'en-US' registered locale.
+  const languageCode = _.first(languageTag.split('-')) || [];
+  return _.find(registeredLocales, (locale) => _.startsWith(locale, languageCode));
+}
+
+export class I18n {
+
+  _registeredTranslations = {};
+
+  constructor(defaultLocale = 'en') {
+    this._defaultLocale = defaultLocale;
+  }
+
+  /**
+   * Return all translations for registered locales
+   * @return {Promise<Object>} translations - A Promise object where keys are
+   *                                          the locale and values are Objects
+   *                                          of translation keys and translations
+   */
+  getAllTranslations() {
+    let localeTranslations = {};
+
+    const locales = this._getRegisteredTranslationLocales();
+    const translations = _.map(locales, (locale) => {
+      return this._getTranslationsForLocale(locale)
+      .then(function (translations) {
+        localeTranslations[locale] = translations;
+      });
+    });
+
+    return Promise.all(translations)
+    .then(translations => _.assign({}, localeTranslations));
+  }
+
+  /**
+   * Return translations for a suitable locale from a user side locale list
+   * @param {Array<Sring>} languageTags -  BCP 47 language tags. The tags are listed in priority order as set in the Accept-Language header.
```

Comment:
This doesn't accept an array of strings, it accepts 1 or more string parameters.